_projects/automate_vulture_whitelist.md

```diff
@@ -0,0 +1,78 @@
+---
+name: "Automate Whitelist creation for vulture"
+collaborating_projects:
+  - coala
+  - vulture
+desc: "Automatically create whitelists and detect false positives for vulture."
+developers_involved:
+  - RJ722
+mentors:
+  - AbdealiJK
+  - jendrikseipp
+  - The-Compiler
+difficulty: medium
+status: in_progress
+initiatives:
+  - GSoC
+issues: []
+markdown: automate_vulture_whitelist.md
+requirements:
+  - "Familiarity with vulture and how whitelists work."
+tags:
+  - Vulture
+  - AST
+  - coverage
+---
+
+The aim of this project is to automatically create whitelists (to be utilized
+by [vulture](https://github.com/jendrikseipp/vulture)) for projects using sip
+files to communicate between C++ level API and Python programs, most
+significant of them being PyQt projects.
+
+#### IDEAS
+
+* Create a script which takes sip files as input and spits a whitelist with all
+  the virtual functions defined in the sip files. (Add Documentation and tests)
+* Use that script for generating a whitelist for popular projects using PyQt
+  (like [qutebrowser](https://github.com/qutebrowser/qutebrowser)).
+* Allow passing coverage.xml to vulture via `--make-whitelist` and let it find
+  the list of false-positive functions. (Add Documentation and tests as well)
+* Write a whitelist generator for shiboken, the binding
+  generator used for PySide2
+
+#### MILESTONES
+
+##### COMMUNITY BONDING
+* Discussion on whether to implement a new Tracer or to directly consume `xml`
+  output from coverage.py
+* Get familiar with coverage.xml
+* Take up small tasks from
+  [TODO](https://github.com/jendrikseipp/vulture/tree/master/TODO.rst)
+* [Follow up milestone](https://gitlab.com/coala/GSoC/GSoC-2018/milestones/28)
+
+##### CODING PHASE 1
+* The script for creating a whitelist from sip files should be implemented.
+* It should reside in it's own repository, with CI enabled for testing on both
+  the GNU/Linux and Windows platforms.
+* It should be gracefully documented.
+* Use this script to create a whitelist for
+  [qutebrowser](https://github.com/qutebrowser/qutebrowser).
+* Look at methods to increase coverage for vulture from it's already excellent
+  95%+ coverage.
+* [Follow up milestone](https://gitlab.com/coala/GSoC/GSoC-2018/milestones/29)
+
+##### CODING PHASE 2
+* `--make-whitelist`: Vulture should be able to find a list of false positives
+  (for functions and methods) from `coverage.xml`.
+* Maintain code coverage, robust tests should be written.
```

Comment:
-> Maintain 100% test code coverage established in coding phase 1.

Reply:
This comment depends on the outcome of this discussion https://github.com/coala/projects/pull/571#discussion_r187789504 ; maybe 100% isnt desirable.